corehq/apps/app_manager/util.py

```diff
@@ -409,7 +409,7 @@ def get_casedb_schema(form):
     This lists all case types and their properties for the given app.
     """
     app = form.get_app()
-    base_case_type = form.get_module().case_type
+    base_case_type = form.get_module().case_type if form.requires_case() else None
```

Comment:
> This looks like a good change, but wondering if it will hide case properties from people who were previously seeing/using them?

I believe the strict answer to this question is "no." `get_casedb_schema` is only used by `get_form_data_schema`, which is only used by vellum to display app properties. Forms that don't require cases don't have the app properties window, so they wouldn't have displayed case properties at all. (I'm actually a bit confused by that - even if usercase in the form builder turned on, the app properties window doesn't seem to show unless the form requires a case. Am I not using user properties in the form builder correctly?)

That said, I do think it's worth considering rolling back or further discussing the case sharing change, because this PR will improve performance for forms that don't require cases, but forms that do require cases have presumably also had a performance hit and they won't benefit from this change.